app/controllers/api/DashboardsApiController.java

```diff
@@ -467,9 +502,8 @@ public Result addWidget(String dashboardId) {
                             widget = new StreamSearchResultCountWidget(dashboard, query, timerange, description, streamId);
                         }
                         break;
-                    }
                     case FIELD_CHART:
-                        Map<String, Object> config = new HashMap<String, Object>() {{
+                        final Map<String, Object> config = new HashMap<String, Object>() {{
```

Comment:
The frontend should provide all those values when creating the widget without the user filling them in, so I think we can use the `ImmutableMap`.